libvips/colour/icc_transform.c

```diff
@@ -720,8 +720,26 @@ vips_icc_set_import( VipsIcc *icc,
 	 */
 	if( code->in &&
 		!icc->in_profile ) {
-		const char *name = code->in->Type == VIPS_INTERPRETATION_CMYK ?
-			"cmyk" : "srgb";
+		const char *name;
+
+		switch( code->in->Type ) {
+		case VIPS_INTERPRETATION_B_W:
+		case VIPS_INTERPRETATION_GREY16:
+			name = "sgrey";
+			break;
+
+		case VIPS_INTERPRETATION_CMYK:
+			name = "cmyk";	
+			break;
+
+		case VIPS_INTERPRETATION_RGB16:
+			name = "p3";
```

Comment:
Is this a good idea? It would meaning converting an RGB image to RGB16 could change the colour drastically.

Reply:
Ah, I forgot to omit this change. This should be part of another PR.

IIUC, this path is only taken for RGB16 -> RGB and not vice versa, but perhaps I misunderstood.